lib/data/dataset.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+//
+// This is an extension of schema.js where we define *datasest-specific* functions
+// for dealing with the XForms XML schema. When a form us uploaded to Central,
+// we check the XML for an <entity> block, which will contain information
+// about datasets and mappings from form fields to
+// dataset properties.
+
+const Option = require('../util/option');
+const Problem = require('../util/problem');
+const { traverseXml, findOne, and, root, node, hasAttr, tree, attr } = require('../util/xml');
+
+const validateDatasetName = (name) => {
```

Comment:
This doesn't handle all the invalid XML identifiers. For example: `×people` (first character is multiplication sign #xd7) is invalid xml identifier but this function returns true.

I would reverse the regex to check valid string and if there's a match then return true.

May be try this: `^[\p{L}_][\p{L}.\d_-]*$. N.B. p{L}`. ref: https://stackoverflow.com/questions/3158274/what-would-be-a-regex-for-valid-xml-names

Reply:
Thanks for the tip! I finally settled on: `/^(?!__)[\p{L}:_][\p{L}:\d_-]*$/u`